egs/hkust/s5b/run.sh

```diff
@@ -0,0 +1,162 @@
+
+# train tri2 [delta+delta-deltas]
+steps/train_deltas.sh --cmd "$train_cmd" \
+ 2500 20000 data/train data/lang exp/tri1_ali exp/tri2 || exit 1;
+
+# decode tri2
+utils/mkgraph.sh data/lang_test exp/tri2 exp/tri2/graph
+steps/decode.sh --cmd "$decode_cmd" --config conf/decode.config --nj 10 \
+  exp/tri2/graph data/dev exp/tri2/decode
+
+# train and decode tri2b [LDA+MLLT]
+
+steps/align_si.sh --cmd "$train_cmd" --nj 10 \
+  data/train data/lang exp/tri2 exp/tri2_ali || exit 1;
+
+steps/align_si.sh --cmd "$train_cmd" --nj 10 \
+  data/train_100k data/lang exp/tri2 exp/tri2_ali_100k || exit 1;
+
+# Train tri3a, which is LDA+MLLT,
+steps/train_lda_mllt.sh --cmd "$train_cmd" \
+ 2500 20000 data/train data/lang exp/tri2_ali exp/tri3a || exit 1;
+
+utils/mkgraph.sh data/lang_test exp/tri3a exp/tri3a/graph || exit 1;
+steps/decode.sh --cmd "$decode_cmd" --nj 10 --config conf/decode.config \
+  exp/tri3a/graph data/dev exp/tri3a/decode
+# From now, we start building a more serious system (with SAT), and we'll
+# do the alignment with fMLLR.
+
+steps/align_fmllr.sh --cmd "$train_cmd" --nj 10 \
+  data/train data/lang exp/tri3a exp/tri3a_ali || exit 1;
+
+steps/train_sat.sh --cmd "$train_cmd" \
+  2500 20000 data/train data/lang exp/tri3a_ali exp/tri4a || exit 1;
+
+utils/mkgraph.sh data/lang_test exp/tri4a exp/tri4a/graph
+steps/decode_fmllr.sh --cmd "$decode_cmd" --nj 10 --config conf/decode.config \
+  exp/tri4a/graph data/dev exp/tri4a/decode
+
+steps/align_fmllr.sh  --cmd "$train_cmd" --nj 10 \
+  data/train data/lang exp/tri4a exp/tri4a_ali
+
+# Building a larger SAT system.
+
+steps/train_sat.sh --cmd "$train_cmd" \
+  3500 100000 data/train data/lang exp/tri4a_ali exp/tri5a || exit 1;
+
+utils/mkgraph.sh data/lang_test exp/tri5a exp/tri5a/graph || exit 1;
+steps/decode_fmllr.sh --cmd "$decode_cmd" --nj 10 --config conf/decode.config \
+   exp/tri5a/graph data/dev exp/tri5a/decode || exit 1;
+
+# MMI starting from system in tri5a.  Use the same data (100k_nodup).
+# Later we'll use all of it.
+steps/align_fmllr.sh --cmd "$train_cmd" --nj 10 \
+  data/train data/lang exp/tri5a exp/tri5a_ali || exit 1;
+
+steps/make_denlats.sh --cmd "$train_cmd" --nj 10 --transform-dir exp/tri5a_ali \
+  --config conf/decode.config \
+  data/train data/lang exp/tri5a exp/tri5a_denlats || exit 1;
+steps/train_mmi.sh --cmd "$train_cmd" --boost 0.1 \
+  data/train data/lang exp/tri5a_ali exp/tri5a_denlats exp/tri5a_mmi_b0.1 || exit 1;
+steps/decode.sh --cmd "$decode_cmd" --nj 10 --config conf/decode.config \
+  --transform-dir exp/tri5a/decode \
+  exp/tri5a/graph data/dev exp/tri5a_mmi_b0.1/decode || exit 1 ;
+
+# Do MPE.
+steps/train_mpe.sh  --cmd "$train_cmd" data/train data/lang exp/tri5a_ali exp/tri5a_denlats exp/tri5a_mpe || exit 1;
+
+steps/decode.sh --cmd "$decode_cmd" --nj 10 --config conf/decode.config \
+  --transform-dir exp/tri5a/decode \
+  exp/tri5a/graph data/dev exp/tri5a_mpe/decode || exit 1 ;
+
+# SGMM system [sgmm5a]
```

Comment:
If you have time, I think it would be better to move the parts of the script that relate to discriminative training and SGMMs, to sub-scripts in local, e.g.
local/run_discriminative.sh
and
local/run_sgmm.sh

and put those commands commented-out in the top-level run.sh, since those aren't the branches of the system building that we particularly recommend at this point.